it/google-cloud-platform/src/test/java/com/google/cloud/teleport/it/gcp/bigtable/BigTableIOLT.java

```diff
@@ -0,0 +1,250 @@
+
+    BigtableIO.Write writeIO =
+        BigtableIO.write().withBigtableOptions(bigtableOptions).withTableId(tableId);
+
+    writePipeline
+        .apply(GenerateSequence.from(0).to(configuration.getNumRows()))
+        .apply(ParDo.of(new MapToBigTableFormat()))
+        .apply("Write to BigTable", writeIO);
+
+    PipelineLauncher.LaunchConfig options =
+        PipelineLauncher.LaunchConfig.builder("test-bigtable-write")
+            .setSdk(PipelineLauncher.Sdk.JAVA)
+            .setPipeline(writePipeline)
+            .addParameter("runner", configuration.getRunner())
+            .build();
+
+    return pipelineLauncher.launch(project, region, options);
+  }
+
+  private PipelineLauncher.LaunchInfo testRead() throws IOException {
+    BigtableIO.Read readIO =
+        BigtableIO.read()
+            .withoutValidation()
+            .withProjectId(project)
+            .withInstanceId(resourceManager.getInstanceId())
+            .withTableId(tableId);
+
+    readPipeline
+        .apply("Read from BigTable", readIO)
+        .apply("Counting element", ParDo.of(new CountingFn<>(READ_ELEMENT_METRIC_NAME)));
+
+    PipelineLauncher.LaunchConfig options =
+        PipelineLauncher.LaunchConfig.builder("test-bigtable-read")
+            .setSdk(PipelineLauncher.Sdk.JAVA)
+            .setPipeline(readPipeline)
+            .addParameter("runner", configuration.getRunner())
+            .build();
+
+    return pipelineLauncher.launch(project, region, options);
+  }
+
+  /** Options for Bigquery IO load test. */
+  @AutoValue
+  abstract static class Configuration {
+    abstract Long getNumRows();
+
+    abstract Integer getPipelineTimeout();
+
+    abstract String getRunner();
+
+    abstract Integer getRowSize();
+
+    static Configuration of(long numRows, int pipelineTimeout, String runner) {
+      return new AutoValue_BigTableIOLT_Configuration.Builder()
+          .setNumRows(numRows)
+          .setPipelineTimeout(pipelineTimeout)
+          .setRunner(runner)
+          .setRowSize(1024)
+          .build();
+    }
+
+    @AutoValue.Builder
+    abstract static class Builder {
+      abstract Configuration.Builder setNumRows(long numRows);
+
+      abstract Configuration.Builder setPipelineTimeout(int timeOutMinutes);
+
+      abstract Configuration.Builder setRunner(String runner);
+
+      abstract Configuration.Builder setRowSize(int rowSize);
+
+      abstract Configuration build();
+    }
+
+    abstract Configuration.Builder toBuilder();
+  }
+
+  /** Maps long number to the BigTable format record. */
+  private static class MapToBigTableFormat extends DoFn<Long, KV<ByteString, Iterable<Mutation>>>
+      implements Serializable {
+
+    @ProcessElement
+    public void processElement(ProcessContext c) {
+      int index = c.element().intValue();
+
+      ByteString key = ByteString.copyFromUtf8(String.format("key%09d", index));
+      ByteString value = ByteString.copyFromUtf8(String.format("value%09d", index));
```

Comment:
The cloud [documentation](https://cloud.google.com/blog/topics/developers-practitioners/bigtable-vs-bigquery-whats-difference) suggests ..."Bigtable can scale to billions of rows and thousands of columns" so this diagonal structured key-value might not be the typical use case for Bigtable

We possibly need a numColumns in Configuration, default set to 1 and make the column key String.format("value%09d", index%numColumn)

Reply:
Not sure if this is the reason of https://github.com/apache/beam/issues/27022. Also going to investigate it

Reply:
I tried to set multiple columns as you suggested, it didn't help with the issue.

Reply:
I find the cause is
- cell does not have set timestamp, so it default to epoch (1970-01-01)
- the createTable has a garbage collection policy of 1h, so large amount data written triggers GC and some records get deleted

We need to use `.setTimestampMicros(java.time.Instant.now().toEpochMilli()*1000)` for Mutation.SetCell

Reply:
Thank you!